packages/strapi-plugin-users-permissions/config/queries/mongoose.js

```diff
@@ -1,14 +1,161 @@
 const _ = require('lodash');
 
+const buildTempFieldPath = field => {
+  return `__${field}`;
+};
+
+const restoreRealFieldPath = (field, prefix) => {
+  return `${prefix}${field}`;
+};
+
+export const generateLookupStage = (
```

Comment:
Same for Mongo. We are writing that kind of generic method in the `strapi-mongoose` hook itself. It avoids code duplication and it's easier to main between query files and generated API.